### ISSUE TITLE
Unable to start Jenkins 2.475

### ISSUE THREAD

**Issue by marvinruder** [jenkinsci/oic-auth-plugin#387]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
Jenkins: 2.474
OS: Linux - 6.9.12-200.fc40.aarch64
Java: 21.0.4 - Eclipse Adoptium (OpenJDK 64-Bit Server VM)
---
ansicolor:1.0.4
antisamy-markup-formatter:162.v0e6ec0fcfcf6
apache-httpcomponents-client-4-api:4.5.14-208.v438351942757
apache-httpcomponents-client-5-api:5.3.1-110.v77252fb_d4da_5
asm-api:9.7-33.v4d23ef79fcc8
authentication-tokens:1.119.v50285141b_7e1
authorize-project:1.7.2
basic-branch-build-strategies:81.v05e333931c7d
bootstrap5-api:5.3.3-1
bouncycastle-api:2.30.1.78.1-248.ve27176eb_46cb_
branch-api:2.1178.v969d9eb_c728e
build-timeout:1.33
caffeine-api:3.1.8-133.v17b_1ff2e0599
checks-api:2.2.1
cloud-stats:336.v788e4055508b_
cloudbees-folder:6.942.vb_43318a_156b_2
command-launcher:115.vd8b_301cc15d0
commons-lang3-api:3.17.0-84.vb_b_938040b_078
commons-text-api:1.12.0-129.v99a_50df237f7
credentials:1371.vfee6b_095f0a_3
credentials-binding:681.vf91669a_32e45
dark-theme:479.v661b_1b_911c01
data-tables-api:2.1.4-1
display-url-api:2.204.vf6fddd8a_8b_e9
docker-commons:443.v921729d5611d
docker-java-api:3.3.6-90.ve7c5c7535ddd
docker-plugin:1.6.2
docker-workflow:580.vc0c340686b_54
dockerhub-notification:2.7.2
durable-task:568.v8fb_5c57e8417
echarts-api:5.5.1-1
eddsa-api:0.3.0-4.v84c6f0f4969e
embeddable-build-status:487.va_0ef04c898a_2
font-awesome-api:6.6.0-1
forensics-api:2.5.0
git:5.4.1
git-client:5.0.0
github:1.40.0
github-api:1.321-475.vf7ef62885c83
github-branch-source:1797.v86fdb_4d57d43
github-checks:583.v9c7da_3c0a_6d5
gson-api:2.11.0-41.v019fcf6125dc
hetzner-cloud:87.v3e433a96543b
instance-identity:185.v303dc7c645f9
ionicons-api:74.v93d5eb_813d5f
jackson2-api:2.17.0-379.v02de8ec9f64c
jakarta-activation-api:2.1.3-1
jakarta-mail-api:2.1.3-1
javax-activation-api:1.2.0-7
javax-mail-api:1.6.2-10
jaxb:2.3.9-1
jjwt-api:0.11.5-112.ve82dfb_224b_a_d
joda-time-api:2.12.7-29.v5a_b_e3a_82269a_
jquery3-api:3.7.1-2
jsch:0.2.16-86.v42e010d9484b_
json-api:20240303-41.v94e11e6de726
json-path-api:2.9.0-58.v62e3e85b_a_655
junit:1296.vb_f538b_c88630
lockable-resources:1301.v0e3b_da_4b_4462
mailer:472.vf7c289a_4b_420
matrix-project:832.va_66e270d2946
mina-sshd-api-common:2.13.2-125.v200281b_61d59
mina-sshd-api-core:2.13.2-125.v200281b_61d59
monitoring:1.99.0
oauth-credentials:0.653.v14cf2088e950
oic-auth:4.324.vfd49d010926b_
okhttp-api:4.11.0-179.vdc1e64343d52
pipeline-build-step:540.vb_e8849e1a_b_d8
pipeline-graph-analysis:216.vfd8b_ece330ca_
pipeline-groovy-lib:730.ve57b_34648c63
pipeline-input-step:495.ve9c153f6067b_
pipeline-milestone-step:119.vdfdc43fc3b_9a_
pipeline-model-api:2.2214.vb_b_34b_2ea_9b_83
pipeline-model-definition:2.2214.vb_b_34b_2ea_9b_83
pipeline-model-extensions:2.2214.vb_b_34b_2ea_9b_83
pipeline-rest-api:2.34
pipeline-stage-step:312.v8cd10304c27a_
pipeline-stage-tags-metadata:2.2214.vb_b_34b_2ea_9b_83
pipeline-stage-view:2.34
plain-credentials:183.va_de8f1dd5a_2b_
plugin-util-api:4.1.0
prism-api:1.29.0-17
resource-disposer:0.23
role-strategy:743.v142ea_b_d5f1d3
scm-api:696.v778d637b_a_762
script-security:1358.vb_26663c13537
simple-theme-plugin:196.v96d9592f4efa_
snakeyaml-api:2.3-123.v13484c65210a_
ssh-agent:376.v8933585c69d3
ssh-credentials:343.v884f71d78167
ssh-slaves:2.973.v0fa_8c0dea_f9f
ssh-steps:2.0.68.va_d21a_12a_6476
sshd:3.330.vc866a_8389b_58
structs:338.v848422169819
theme-manager:262.vc57ee4a_eda_5d
timestamper:1.27
token-macro:400.v35420b_922dcb_
trilead-api:2.147.vb_73cc728a_32e
variant:60.v7290fc0eb_b_cd
workflow-aggregator:600.vb_57cdd26fdd7
workflow-api:1336.vee415d95c521
workflow-basic-steps:1058.vcb_fc1e3a_21a_9
workflow-cps:3953.v19f11da_8d2fa_
workflow-durable-task-step:1371.vb_7cec8f3b_95e
workflow-job:1436.vfa_244484591f
workflow-multibranch:795.ve0cb_1f45ca_9a_
workflow-scm-step:427.v4ca_6512e7df1
workflow-step-api:678.v3ee58b_469476
workflow-support:920.v59f71ce16f04
ws-cleanup:0.46
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Container `alpine-jdk21` (Alpine Linux v3.20.2) on host Fedora CoreOS 40.20240808.3.0

### Reproduction steps

1. Upgrade container to 2.475 and start
2. Observe the following log when starting the container:

```
2024-09-05 07:47:41.362+0000 [id=42]	WARNING	jenkins.model.Jenkins$5#runTask: Loading global config failed perhaps due to plugin dependency issues
java.lang.NoSuchMethodError: 'void hudson.security.ChainedServletFilter.<init>(javax.servlet.Filter[])'
	at PluginClassLoader for oic-auth//org.jenkinsci.plugins.oic.OicSecurityRealm.createFilter(OicSecurityRealm.java:828)
```

3. Observe the following log when opening Jenkins web interface, which shows 500 error page

```
2024-09-05 07:48:07.730+0000 [id=28]	WARNING	h.i.i.InstallUncaughtExceptionHandler#handleException: Caught unhandled exception with ID f8a76819-f75b-4002-8317-2a10b3d5a940
hudson.security.AccessDeniedException3: anonymous is missing the Overall/Read permission
	at hudson.security.ACL.checkPermission(ACL.java:80)
```

### Expected Results

Jenkins starts up normally and users can log in.

### Actual Results

Jenkins shows 500 error page and login is not possible

### Anything else?

Jenkins changelog holds the following information for 2.475:

> Upgrade Spring Framework from 5.3.39 to 6.1.12, upgrade Spring Security from 5.8.14 to 6.3.3, and upgrade Java EE from 8 to 9. Users of the LDAP plugin must upgrade it to version 733.vd3700c27b_043 in lockstep with upgrading Jenkins core. Users of third-party servlet containers must upgrade the servlet container to an EE 9 version in accordance with the Jenkins Servlet Container Support Policy. [Spring Framework 6.0.23 release notes](https://github.com/spring-projects/spring-framework/releases/tag/v6.0.23), [Spring Framework 6.1.12 release notes](https://github.com/spring-projects/spring-framework/releases/tag/v6.1.12), [Spring Security 6.2.6 release notes](https://github.com/spring-projects/spring-security/releases/tag/6.2.6), [Spring Security 6.3.2 release notes](https://github.com/spring-projects/spring-security/releases/tag/6.3.2), [Spring Security 6.3.3 release notes](https://github.com/spring-projects/spring-security/releases/tag/6.3.3), [Jarkata EE 9 release page](https://projects.eclipse.org/releases/jakarta-ee-9), [LDAP plugin 733.vd3700c27b_043](https://github.com/jenkinsci/ldap-plugin/releases/tag/733.vd3700c27b_043), [Servlet Container Support Policy](https://www.jenkins.io/doc/book/platform-information/support-policy-servlet-containers/)

**Comment by basil** [comment]
Fixed in https://github.com/jenkinsci/jenkins/pull/9696

**Comment by basil** [comment]
Released in 2.476. If issues are still encountered with the 2.476 weekly release, file a _separate_ ticket with the full exception stack trace and details about how to reproduce the issue from scratch.
